util/json.go

```diff
@@ -41,6 +58,11 @@ func NewJSONDecoder(r io.Reader) *json.Decoder {
 func MustUnmarshalJSON(bs []byte) interface{} {
 	var x interface{}
 	if err := UnmarshalJSON(bs, &x); err != nil {
+		// If the type of err is MalformedJSONInputError, the data is decoded.
+		// In this case, this func should not be panicked.
+		if _, ok := err.(MalformedJSONInputError); ok {
+			return x
+		}
```

Comment:
I'm not sure about this, it seems like it should be considered an error still. I think the assumption here is that the bytes given must be valid json, as-is something like `{"k":1}{!{!!!!!` would be OK, which IMO should be an error.

Reply:
I fixed it to be panicked, but other tests are failed because of that 🤔 
Why `topdown_test.go` passes every expected value into `MustUnmarshalJSON`?
It's weird because some of the value obviously invalid as a JSON(like foo), but still succeeded before the fix.
I want to know what kind of values are expected and how to fix it.

Reply:
I want to fix it, but this test method is tightly-coupled with unpredictable input,
so I workaround by using `UnmarshalJSON` for that and left a comment 😢
If it's not acceptable, I'll try, but I think it will take a long time to fix.

Reply:
Interesting.. 🤔 I'm wondering what those tests were expecting to happen before if they're passing in things that would start to fail now.

I'll take a look. We will probably want to sort this out before merging the changes as this affects a lot of places, but I don't think we need to change the implementation here. It seems like we are now doing the right thing when decoding.

Reply:
Ok, I did a quick pass over them and I think we're in good shape. This was turning up *real* errors where the stuff we passed in was just bad JSON. It was only in a handful of tests.

The required changes to fix them are here: https://github.com/open-policy-agent/opa/commit/383ccdaef31eb828ece1a0d7f5e15f8a4a6923fe Please include them with the PR and undo the changes in `topdown_test.go`. Then we should be good to go.

Reply:
Thanks for the fix!
I cherry-picked your commit and erase FIXME comment 😄 
Please review it again.